app/src/main/java/com/x8bit/bitwarden/data/vault/datasource/sdk/model/Fido2CredentialRegistrationUserInterfaceImpl.kt

```diff
@@ -0,0 +1,40 @@
+package com.x8bit.bitwarden.data.vault.datasource.sdk.model
+
+import com.bitwarden.fido.CheckUserOptions
+import com.bitwarden.sdk.CheckUserResult
+import com.bitwarden.sdk.CipherViewWrapper
+import com.bitwarden.sdk.Fido2UserInterface
+import com.bitwarden.sdk.UiHint
+import com.bitwarden.vault.CipherView
+import com.bitwarden.vault.Fido2CredentialNewView
+import com.x8bit.bitwarden.data.platform.annotation.OmitFromCoverage
+
+/**
+ * Implementation of [Fido2UserInterface] for registering new FIDO 2 credentials.
+ */
+@OmitFromCoverage
+class Fido2CredentialRegistrationUserInterfaceImpl(
+    private val isVerificationSupported: Boolean,
+    private val checkUser: suspend (CheckUserOptions, UiHint?) -> CheckUserResult,
+    private val checkUserAndPickCredentialForCreation: suspend (
+        options: CheckUserOptions,
+        newCredential: Fido2CredentialNewView,
+    ) -> CipherViewWrapper,
+) : Fido2UserInterface {
+
+    override suspend fun checkUser(
+        options: CheckUserOptions,
+        hint: UiHint,
+    ): CheckUserResult = checkUser.invoke(options, hint)
+
+    override suspend fun checkUserAndPickCredentialForCreation(
+        options: CheckUserOptions,
+        newCredential: Fido2CredentialNewView,
+    ): CipherViewWrapper = checkUserAndPickCredentialForCreation.invoke(options, newCredential)
+
+    override suspend fun isVerificationEnabled(): Boolean = isVerificationSupported
+
+    override suspend fun pickCredentialForAuthentication(
+        availableCredentials: List<CipherView>,
+    ): CipherViewWrapper = throw IllegalStateException()
```

Comment:
@fedemkr I extracted `Fido2CredentialStore`, but even with concrete implementations of `Fido2UserInterface`, we're still going to end up throwing an exception here because we know `pickCredentialForAuthentication` should never be invoked in this process. 

Alternatively, if we create a single `Fido2UserInterface` implementation we have to pass arguments to `registerFido2Credential` that are unrelated to the operation. 

```kotlin
fun registerFido2Credential(
    pickCredentialForAuthentication: (List<CipherView>) -> CipherViewWrapper,
    // other params omitted for brevity
)
```
☝️ that just doesn't seem right to me. 

I think creating two different `Fido2UserInterface` implementations that `throw` if unexpected delegates are invoked is a fair middle-ground. Throwing an exception at least acts as a safeguard against the delegate unexpectedly firing. Taking this approach ensures we gracefully shut down the process and notify the user of a failure instead of putting them in a potentially weird or vulnerable state.

Reply:
Yep, sounds good 😄 Thanks for looking into it!